src/app/lab/lab.component.ts

```diff
@@ -24,22 +24,32 @@ import * as _ from 'lodash';
 export class LabComponent implements OnInit, OnDestroy {
 
     private routeParamsSubscription: Subscription;
-    public project: Project;
+    public project: APIProject;
 
     constructor(private projectService: ProjectService,
                 private route: ActivatedRoute,
                 private router: Router) {
     }
 
+    // Save
+    @HostListener('window:beforeunload', ['$event'])
+    unloadHandler(event) {
+        this.saveChartSettings();
+    }
+
     ngOnInit() {
         // Load existing project id or redirect to dashboard
         this.routeParamsSubscription = this.route.params.subscribe(params => {
             let id: string = params['id'];
             if (id !== undefined) {
-              this.project = this.projectService.get(id);
-            }
-            // Reroute to dashboard if project doesn't exist
-            if (!this.project) {
+              this.projectService.get(id).subscribe(data => {
+                    this.project = data;
+                    if (!this.project) {
```

Comment:
If a project id doesn't exist, the API will throw a 404 which is handled by the observer `catch` method rather than the success method. I wouldn't expect data to ever be null/undefined here. Do the router navigate in the catch handler instead.